pennylane_lightning/lightning_tensor/lightning_tensor.py

```diff
@@ -0,0 +1,358 @@
+# limitations under the License.
+"""
+This module contains the LightningTensor class that inherits from the new device interface.
+It is a device to perform tensor network operations on a quantum circuit. 
+"""
+from dataclasses import replace
+from numbers import Number
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pennylane as qml
+from pennylane.devices import DefaultExecutionConfig, Device, ExecutionConfig
+from pennylane.devices.modifiers import simulator_tracking, single_tape_support
+from pennylane.tape import QuantumTape
+from pennylane.transforms.core import TransformProgram
+from pennylane.typing import Result, ResultBatch
+
+from .quimb._mps import QuimbMPS
+
+Result_or_ResultBatch = Union[Result, ResultBatch]
+QuantumTapeBatch = Sequence[QuantumTape]
+QuantumTape_or_Batch = Union[QuantumTape, QuantumTapeBatch]
+PostprocessingFn = Callable[[ResultBatch], Result_or_ResultBatch]
+
+
+_backends = frozenset({"quimb"})
+# The set of supported backends.
+
+_methods = frozenset({"mps"})
+# The set of supported methods.
+
+
+def accepted_backends(backend: str) -> bool:
+    """A function that determines whether or not a backend is supported by ``lightning.tensor``."""
+    return backend in _backends
+
+
+def accepted_methods(method: str) -> bool:
+    """A function that determines whether or not a method is supported by ``lightning.tensor``."""
+    return method in _methods
+
+
+@simulator_tracking
+@single_tape_support
+class LightningTensor(Device):
+    """PennyLane Lightning Tensor device.
+
+    A device to perform tensor network operations on a quantum circuit.
+
+    Args:
+        wires (int): The number of wires to initialize the device with.
+            Defaults to ``None`` if not specified.
+        backend (str): Supported backend. Currently, only ``quimb`` is supported.
+        method (str): Supported method. Currently, only ``mps`` is supported.
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Currently, it can only be ``None``, so that computation of
+            statistics like expectation values and variances is performed analytically.
+        c_dtype: Datatypes for the tensor representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        **kwargs: keyword arguments. The following options are currently supported:
+
+            ``max_bond_dim`` (int): Maximum bond dimension for the MPS simulator.
+                It corresponds to the number of Schmidt coefficients retained at the end of the SVD algorithm when applying gates. Default is `None`.
+            ``cutoff`` (float): Truncation threshold for the Schmidt coefficients in a MPS simulator. Default is `1e-16`.
+            ``return_tn`` (bool): Whether to return the tensor network object along with the results of circuit execution. Default is `False`.
```

Comment:
What is the TN object type? Some sort of a graph?

I'm not convinced that this makes sense to be part of the return due to questions around the return shape.
When ``return_tn`` is ``True``, is the proposal to return a tuple with the TN object at the end?

Reply:
This is a good point. I took this from the requirements in the PRD (the same goes for ``rehearse`` parameter below) and I am implementing this in the following PR. However, I also think that the tensor should not be part of the return type of the `execution` function (it seems to me that it violates the principles of the new device API). If we must insert this, maybe it makes more sense to add an attribute to the `LightningTensor` class that stores the wavefunction in a dense representation (so that it does not depend on the specific interface) if ``return_tn`` is `True`

Reply:
Agreed, let's leave out `return_tn` and `rehearse` from the initial version. Would you mind creating stories in Shortcut to track our intention to revisit these questions?